### PR TITLE
Convert AI-Hearing into CBA-Setting

### PR DESCRIPTION
**Pull request by Dorbedo**
* moves AI_Hearing from optional into the main
* adds a CBA-Setting
* disabled by default

### CODE REVIEW

File: addons/ai_hearing/XEH_PreInit.sqf [hunk 1]
```diff
@@ -0,0 +1,17 @@
+#include "script_component.hpp"
+
+#include "XEH_PREP.sqf"
+
+[
+    "TFAR_AICanHearPlayer",
+    "CHECKBOX",
+    [LSTRING(SETTING_HEADER), LSTRING(SETTING_DESC)],
+    "Task Force Arrowhead Radio",
```
Comment (dedmen) [inline addons/ai_hearing/XEH_PreInit.sqf]: Don't we have a stringtable entry for that?

File: addons/ai_hearing/functions/fnc_initializeAIHearing.sqf [hunk 2]
```diff
@@ -3,21 +3,25 @@
...
 */
 
-if (!hasInterface) exitWith {}; //Only on clients
+if ((!hasInterface)||{!TFAR_AICanHearPlayer}||{missionNamespace getVariable [QGVAR(isInitialized),false]}) exitWith {}; //Only on clients
+
+GVAR(isInitialized) = true;
+
 
 ["TFAR_AI_Detection", "OnSpeak", {
+    If !(TFAR_AICanHearPlayer) exitWith {};
```
Comment (dedmen) [inline addons/ai_hearing/functions/fnc_initializeAIHearing.sqf]: I would rather just have the setting call removeEventhandler when it get's disabled.

Comment (dedmen) [reply addons/ai_hearing/functions/fnc_initializeAIHearing.sqf]: But this event only fires on the player so.. Probably doesn't really matter if we have an additional check here.
But Setting should still removeEH if GVAR(isInitialized)

Comment (Dorbedo) [reply addons/ai_hearing/functions/fnc_initializeAIHearing.sqf]: I would not recomend this to be enabled and disabled during the mission anyways.
I'll change it.

File: addons/ai_hearing/XEH_PreInit.sqf (hunk 1 above)
Comment (Dorbedo) [reply addons/ai_hearing/XEH_PreInit.sqf]: This is the not localized Category Name.

Comment (dedmen) [reply addons/ai_hearing/XEH_PreInit.sqf]: Welll.. On the other end it's a Name.
Names usually don't get translated

**Comment by Dorbedo** [comment]
done

**Comment by dedmen** [comment]
I'll leave it open till I'm ready to make the next pre-release

**Comment by dedmen** [comment]
Using the CBA eventhandler also fixes https://github.com/michail-nikolaev/task-force-arma-3-radio/issues/1300
Which was caused because the old one checked against the `player` from before respawn.

**Pull request merged**